libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/PropertyMocks/PropertyMock.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Dialogs.Debugging;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.PropertyMocks
+{
+    public abstract class PropertyMock
```

Comment:
It would be nice to have the ability to set user and conversation properties on the first turn of the test.
* "user.foo" = 13
* "conversation.blat.y" = "hello"
etc.
I think the way to do that would to change the rootDialog for a testscript like this:

RootDialog = new AdaptiveDialog()
{  
    Actions = [  
        new SetProperties() { ... array of property/value pairs from the mock.... }
        ReplaceDialog(realRootDialog}
    ]
}

Reply:
Good idea! I look for ways to use DialogStateManager but it requires DialogContext..